File: firebase-api-library/src/main/java/com/fazemeright/firebase_api_library/api/DatabaseStore.java
package com.fazemeright.firebase_api_library.api;

import androidx.annotation.Nullable;
import com.fazemeright.firebase_api_library.listeners.OnTaskCompleteListener;
import java.util.List;
import java.util.Map;
import javax.annotation.Nonnull;

public interface DatabaseStore {

  void storeUserData(@Nonnull String uid, @Nonnull Map<String, Object> userData);

  void storeMessage(@Nonnull Map<String, Object> messageHashMap, String currentUserUid);

  void getAllMessagesForUser(@Nonnull String currentUserUid,
                             @Nullable
                                 OnTaskCompleteListener<List<Map<String, Object>>> listener);
}

File: firebase-api-library/src/main/java/com/fazemeright/library/api/domain/database/DatabaseStore.kt
package com.fazemeright.library.api.domain.database

import com.fazemeright.library.api.Storable
import com.fazemeright.library.api.result.Result

interface DatabaseStore {
    suspend fun storeUserData(uid: String, userData: Storable): Result<Boolean>
    suspend fun storeMessage(storable: Storable, currentUserUid: String): Result<Boolean>
    suspend fun getAllMessagesForUser(currentUserUid: String): Result<List<Map<String, Any>>>?
}